tests/main/searching/task.yaml

```diff
@@ -1,25 +1,22 @@
 summary: Check snap search
 
 execute: |
-    echo "List all snaps"
-    expected="(?s)Name +Version +Developer +Notes +Summary *\n\
-    [a-z-]+ .*?\
-    .*"
-    snap find | grep -Pzq "$expected"
+    echo "Try to list all snaps"
+    ( snap find 2>&1 || echo FAILED ) | grep -Pzq "(?ms)empty query.*FAILED"
```

Comment:
`-z` means I can apply a single regular expression to the whole string. `(?X)` means turn on option X from that point on in the regular expression. Option `m` means to treat the string as multiple lines, so you can have `^` and `$` match just after and before a `\n` inside a single string (I think grep doesn't support this). `s` means to treat the string as a single line, meaning to change `.` to match `\n`. They're usually set together.